File: Codepractice15.py
from collections import Counter

class Solution:
    def lexGreaterPermutation(self, s: str, target: str) -> str:
        n = len(s)
        count = Counter(s)

        for i in range(n - 1, -2, -1):
            cur_count = count.copy()
            possible = True
            for j in range(i):
                if cur_count[target[j]] > 0:
                    cur_count[target[j]] -= 1
                else:
                    possible = False
                    break
            
            if not possible:
                continue

            next_char = None
            if i >= 0:
                for c in range(ord(target[i]) - ord('a') + 1, 26):
                    ch = chr(ord('a') + c)
                    if cur_count[ch] > 0:
                        next_char = ch
                        break
                if next_char is None:
                    continue

            res = []
            for j in range(i):
                res.append(target[j])
            
            if i >= 0:
                res.append(next_char)
                cur_count[next_char] -= 1

            for c in range(26):
                ch = chr(ord('a') + c)
                if cur_count[ch] > 0:
                    res.append(ch * cur_count[ch])

            ans = "".join(res)
            if i == -1:
                if ans > target:
                    return ans
            else:
                return ans

        return ""

File: Codepractice15.java
public class Solution {
    public String lexGreaterPermutation(String s, String target) {
        int n = s.length();
        int[] count = new int[26];
        for (char c : s.toCharArray()) {
            count[c - 'a']++;
        }

        for (int i = n - 1; i >= -1; --i) {
            int[] curCount = count.clone();
            boolean possible = true;
            for (int j = 0; j < i; ++j) {
                if (--curCount[target.charAt(j) - 'a'] < 0) {
                    possible = false;
                    break;
                }
            }
            if (!possible) continue;

            int nextChar = -1;
            if (i >= 0) {
                for (int c = target.charAt(i) - 'a' + 1; c < 26; ++c) {
                    if (curCount[c] > 0) {
                        nextChar = c;
                        break;
                    }
                }
                if (nextChar == -1) continue;
            }

            StringBuilder res = new StringBuilder();
            for (int j = 0; j < i; ++j) {
                res.append(target.charAt(j));
            }
            if (i >= 0) {
                res.append((char) ('a' + nextChar));
                curCount[nextChar]--;
            }

            for (int c = 0; c < 26; ++c) {
                while (curCount[c] > 0) {
                    res.append((char) ('a' + c));
                    curCount[c]--;
                }
            }

            String ans = res.toString();
            if (i == -1) {
                if (ans.compareTo(target) > 0) return ans;
            } else {
                return ans;
            }
        }

        return "";
    }
}